### DISCUSSION TITLE
How can change the color of the meshes to transparent colors?

### DISCUSSION THREAD

**Discussion by Mehrshad-Ghorbanzadeh** [jimy-byerley/pymadcad#20] [Q&A] [answered]
Hello and thanks for your extraordinary effort. I had some questions as follows:
Is it possible to turn 3D mesh colors into transparent colors? 
The second question is that possible to create a grid for the X and Y plan?
The last question is how can add the symbols x, y, and z next to the corresponding axis for the coordinate axes?
In advance, thanks for your guidance.

![1111](https://user-images.githubusercontent.com/53825204/150138062-dcc0cae2-73d5-4f78-9e6b-3a5e05c68b7b.PNG)

**Comment by jimy-byerley** [comment, marked as answer]
Hello there

- Is it possible to turn 3D mesh colors into transparent colors?

    Depending on what exactly you mean. 
    You can easily enable a display mode were meshes are shown transparent, but this more like a sort of wireframe mode. 
    You cannot make a mesh renders half transparent but still having reflects, shading and so, and this is intentional
   
    This intentional for reasons regarding the openGL pipeline:  just setting the surface as transparent can lead many visual artifacts than give inconsistent rendering, and to fix that is highly decreasing the framerate (Also that's the same story in video game rendering pipeline, you may have already seen things like these:
![teapot with transparency artifacts](https://user-images.githubusercontent.com/7940753/150434891-156f1bab-3b5c-4361-becd-6b84b1adc1fb.png)

   The wireframe-like transparency mode resembles to this: 
    ![shaded part](https://user-images.githubusercontent.com/7940753/150435291-824b41d9-7902-4c6c-aac4-bef2328c3268.png)

    ```python
    show([mypart], options={'display_faces':False})
    ```

    ![wireframe part](https://user-images.githubusercontent.com/7940753/150435322-5ff0b2d7-0723-425f-ac5e-cef49f7858f1.png)

   Is that sufficient for your need ?

- The second question is that possible to create a grid for the X and Y plan?

   I made me a Display object to show an infinite grid like this, but I'm using it to display a grid in the view plane, not a preset plane. It's easy to modify it for your need though. the existing object is [`madcad.displays.GridDisplay`](https://github.com/jimy-byerley/pymadcad/blob/e809a1a3a52d67c27c562bf7c1a80022634a2cdb/madcad/displays.py#L648)

    ```python
    from madcad.rendering import Displayable
    from madcad.displays import GridDisplay
    show([ 
            myobject,  
            Displayable(GridDisplay, center=vec3(0)),
            ])
    ```
   It shows something like this
    ![image](https://user-images.githubusercontent.com/7940753/150437054-a9356a27-5c7f-4648-ba3e-87a030140e94.png)


- The last question is how can add the symbols x, y, and z next to the corresponding axis for the coordinate axes?

    I'm afraid there is no choice but to put text annotations next to the axis or to the points placed on those directions. So like it seems your are doing already.
    Using `madcad.text.Text` or `madcad.scheme.note_floating`

I hope these answers are of any help

**Comment by Mehrshad-Ghorbanzadeh** [comment]
Thank you so much. I have to work on it more.

**Reply by jimy-byerley** [reply]
I wonder if I haven't been a bit tough when I made my decision that the surfaces will not allow transparency. I can eventually change this to allow the faces to be transparent, just that transparency artifact will appear ... and it would be up the the user to decide.

**Reply by Mehrshad-Ghorbanzadeh** [reply]
Thank you for your effort. I hope this will be possible in future versions.
